File: Antlr/src/main/scala/io/github/chaosunity/casc/parsing/expression/ConstructorCall.scala
package io.github.chaosunity.casc.parsing.expression

import com.google.common.collect.Lists
import io.github.chaosunity.casc.parsing.`type`.{ClassType, Type}

import java.util

class ConstructorCall(val className: String,
                      arg: util.List[Expression]) extends Call {
    def this(identifier: String) {
        this(identifier, Lists.newArrayList())
    }

    override def arguments: util.List[Expression] = arg

    override def identifier: String = className

    override def `type`: Type = new ClassType(className)

    override def negative: Boolean = false
}

File: src/main/kotlin/io/github/chaosunity/casc/parsing/node/expression/ConstructorCall.kt
package io.github.chaosunity.casc.parsing.node.expression

import io.github.chaosunity.casc.parsing.type.ClassType
import io.github.chaosunity.casc.parsing.type.Type

data class ConstructorCall(private val className: String, override val arguments: List<Argument> = listOf()) :
    Call<ConstructorCall> {
    override val identifier: String = className
    override val type: Type = ClassType(className)
}